### PR TITLE
Add logging to determine why variable tests were timing out

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4047

### CODE REVIEW

File: src/client/common/process/baseDaemon.ts [hunk 1]
```diff
@@ -177,7 +178,7 @@ export abstract class BasePythonDaemon {
         return Object.keys(options).every((item) => daemonSupportedSpawnOptions.indexOf(item as any) >= 0);
     }
     protected sendRequestWithoutArgs<R, E, RO>(type: RequestType0<R, E, RO>): Thenable<R> {
-        if (this.proc && typeof this.proc.exitCode !== 'number') {
+        if (this.isAlive && this.proc && typeof this.proc.exitCode !== 'number') {
```
Comment (DonJayamanne) [inline src/client/common/process/baseDaemon.ts:181]: While running the tests locally found that there were some other unhandled exceptions.
Basically after the tests we kill the daemon, however some async code is still attempting to write to the daemon.

This ensures we handle that situation & thus prevent logging or unhandled errors (i.e. spurious errors being logged)

File: src/test/datascience/debugger.functional.test.tsx [hunk 2]
```diff
@@ -155,8 +158,9 @@ suite('DataScience Debugger tests', () => {
         }
 
         // Start the jupyter server
+        traceInfoIf(!!process.env.VSC_CI_ENABLE_TOO_MUCH_LOGGING, 'Getting interactive window');
         const history = await getOrCreateInteractiveWindow(ioc);
-
+        traceInfoIf(!!process.env.VSC_CI_ENABLE_TOO_MUCH_LOGGING, 'Got interactive window');
```
Comment (DonJayamanne) [inline src/test/datascience/debugger.functional.test.tsx:163]: Added some logging to find out what part is hanging when the variable tests timeout.

File: src/test/datascience/testHelpers.tsx [hunk 3]
```diff
@@ -59,29 +59,41 @@ async function testInnerLoop(
 
 export function runDoubleTest(
     name: string,
-    testFunc: (type: 'native' | 'interactive') => Promise<void>,
+    testFunc: (this: Mocha.Context, type: 'native' | 'interactive') => Promise<void>,
```
Comment (DonJayamanne) [inline src/test/datascience/testHelpers.tsx:62]: Fixes to properly pass the context/scope, so we can use `this.skip`.
We had `c.skip` but that didn't work as the scope wasn't setup properly.

File: src/test/initialize.ts [hunk 4]
```diff
@@ -76,23 +76,43 @@ export async function closeActiveNotebooks(): Promise<void> {
...
-                reject(ex);
-            }
-        );
-    });
+    class CloseEditorsTimeoutError extends Error {
+        constructor() {
+            super("Command 'workbench.action.closeAllEditors' timed out");
+        }
+    }
+    const closeWindowsImplementation = (timeout = 15_000) => {
```
Comment (DonJayamanne) [inline src/test/initialize.ts:84]: Found in the PR that VS Code was timing out here.
We've had this issue in the past hence the timeout.
Updated to ensure we retry again after sleeping for 1/2s.
Possible VSC is stuck/busy, hence giving it time & retrying.

I.e. a work around for something flaky in VS Code.

**Review by rchiodo — APPROVED** [review]

File: src/client/common/process/baseDaemon.ts (hunk 1 above)
Comment (IanMatthewHuff) [reply src/client/common/process/baseDaemon.ts:181]: Nice. Think that I'd seen those errors cluttering up the logs before.

**Review by IanMatthewHuff — APPROVED** [review]

File: .github/workflows/pr.yml [hunk 5]
```diff
@@ -600,6 +600,20 @@ jobs:
 
       # Used by tests for non-python kernels.
       # Test are enabled via env variable `VSC_JUPYTER_CI_RUN_NON_PYTHON_NB_TEST`
+      - name: Install Dot.net
```
Comment (DonJayamanne) [inline .github/workflows/pr.yml:603]: This seems to have been removed from main, I added it back.

File: .github/workflows/pr.yml [hunk 6]
```diff
@@ -338,20 +338,20 @@ jobs:
...
-      - name: Install Dot.net	
-        uses: actions/setup-dotnet@v1	
-        with:	
-          dotnet-version: ${{env.DOTNET_VERSION}}	
-        if: matrix.test-suite != 'notebookWithoutPythonExt'	
 
-      - name: Install .NET Interactive	
-        run: dotnet tool install -g --add-source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" Microsoft.dotnet-interactive	
-        if: matrix.test-suite != 'notebookWithoutPythonExt'	
+      - name: Install Dot.net
```
Comment (DonJayamanne) [inline .github/workflows/pr.yml:342]: Not sure why this is here, but leave it (no harm cause).
Difficult with all of the copy paste going on in github actions - problem is with github actions.

**Pull request merged**